### DISCUSSION TITLE
SolidJS support

### DISCUSSION THREAD

**Discussion by ac130kz** [remix-run/remix#578] [Proposals]
Any plans to support other React-like frameworks? SolidJS has a close to the React's API, but it performs a lot better, which positively affects user experience. In case SolidJS support is not possible in the official way, can anyone tell whether porting Remix to SolidJS is a feasible idea?

**Comment by tigerabrodi** [comment]
@ac130kz 

I have spoken about this with @ryansolid :+1: 

It may be possible to get SolidJS support, but it'll require a lot of work.

React Router and remix-react must both be rewritten into SolidJS as the first step.... :sweat_smile: 

I had many thoughts, and badly wanted to do this, but I think when it comes to Solid, I rather wait for Solid Start if any sort of fullstack solutions with solid.

**Reply by sergiodxa** [reply]
You probably want to wait until the team makes part of the React package independent of React, for example the transition management, then building wrapper for Solid or Vue or anything else should be easier

**Reply by tigerabrodi** [reply]
@sergiodxa ye makes more sense hahaha :smile: 

Cuz if it is hard, we out boss :joy: :handshake:

**Reply by devinrhode2** [reply]
In @ryancarniato's "Solid.js: Past, Present and Future" stream: https://www.youtube.com/watch?v=dOgnQ9SuKYk&t=7555s

It sounds like Mr Carniato is working on some serialization primitives, essentially taking ideas from Remix and Qwik. He said he has something that's kind of like react-query meets Remix

**Comment by buzzeins** [comment]
Or just consider implementing Atro into Remix. Besides React, Astro has also Solid, Svelte, Vue, Alpine, Lit and Preact build in.

https://github.com/withastro/rfcs/discussions/70
https://astro.build/

Also the Partial Hydration concept in Remix would be a bigdeal, isn't it? ;)

**Comment by contactjavas** [comment]
Next 13, Vercel, and RSC are the hot stuff on my Twitter feed.
It may be time for Remix to support SolidJS.

**Comment by devinrhode2** [comment]
SolidStart is basically Remix, you can disable javascript, uses the same router data loading patterns, also has some `key` api similar to `react-query` which you can use to invalidate cached data.

I'm EXTREMELY impressed by SolidStart. I highly recommend trying it on a little project if you can!

**Comment by christhornham** [comment]
> SolidStart is basically Remix

@devinrhode2  It sounds like you have experience with both Solidstart and Remix. Is that correct?

If so... 

I'm relatively new to the frontend world. I really like Solid, but there aren't many in-depth tutorials on the subject. I learned Solid by taking a React course and converting the entire thing to Solid by digging through the docs. That was great because I got to learn both React and Solid at the same time.

I'm now trying to learn how meta frameworks like Next, Remix, Nuxt, Sveltekit, Solidstart etc work. I don't fully understand SSR, mixing frontend code with backend code etc. 

Here's my question.

1. If you have experience with both Remix and Solidstart, do you think I could learn about Solidstart by taking a Remix course and converting it to Solidstart? 

To be clear, I don't expect Remix to be exactly like Solidstart. I'm wondering if the concepts in Remix are similar enough to learn  Solidstart.

**Reply by devinrhode2** [reply]
The key thing Remix has pioneered is making the Server-Side Rendered html fully interactive. I believe I have actually had many situations where hydration screws up my input experience, namely the top search box of a site, but could also be the "Add to Cart" button, etc.

SolidStart, Qwik, and Svelte have all copied this "fully usable initial html" approach.

Theres another benefit: if your backend has issues, or you are migrating to a new technology, as long as you can continue serving cached html and handling the form POST requests, your app is effectively running, albeit SLIGHTLY  slower for users with a fast connection and plenty of cpu/ram.

To answer your question, learning Remix is a great idea not only if you want to learn SolidStart, but if you want a step ahead for any of these next-gen frameworks.

The only caveat is that I'm not sure if Next.js 14 supports the "fully usable initial html" feature of Remix. (Also, the more common term for "fully usable initial html" is a "progressive web app", but I feel that term isn't descriptive enough)

**Reply by christhornham** [reply]
That’s great information. Thanks for the detailed answer!

**Reply by devinrhode2** [reply]
Tanstack Router is the spiritual successor to use these days, or Tanstack Start. Solid.js is different enough that the learning resources will always be limited

**Comment by brophdawg11** [comment]
This discussion is being closed as part of the Remix v3 repository cleanup.

Remix v1/v2 was [upstreamed into React Router v7](https://remix.run/blog/wake-up-remix), so if this proposal is still relevant, please reopen or continue the discussion in the [React Router](https://github.com/remix-run/react-router) repository.

**Reply by devinrhode2** [reply]
Keep it closed. Shoulda been closed long time ago
